### PR TITLE
Fix TaskProcessorFactory#getTaskProcessor get common processor is not thread safe

### LINKED ISSUE #10470: [Bug] [Master] ProcessInstance status may incorrect  in high pressure.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

When I test on my laptop, I find the process instance will still in running status,
image

and the task will not be dispatched.
image

### What you expected to happen

The process instance can be executed success.

### How to reproduce

Insert 10000 command in to t_ds_command, and wait the command be executed.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

Close #10470, maybe we need to remove the variable in processor, then we don't need to create it in each task, or we can use threadlocal to store the variable.

## Brief change log

- Don't cache the CommonTaskProcessor in TaskProcessorFactory
- Add some log
## Verify this pull request

Trigger multiple process instance, then see all process instance can run successful.

### CODE REVIEW

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/ProcessInstance.java [hunk 1]
```diff
@@ -265,12 +265,13 @@ public ProcessInstance() {
...
+        // todo: the name is not unique
         this.name = processDefinition.getName()
-            + "-"
-            +
-            processDefinition.getVersion()
-            + "-"
-            +
+                + "-"
+                +
+                processDefinition.getVersion()
```
Comment (kezhenxu94) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/ProcessInstance.java]: nit: what about also simplify this with `String.join` or `Joiner` in guava, it should be improved in readability?

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/consumer/TaskPriorityQueueConsumer.java [hunk 2]
```diff
@@ -190,16 +190,18 @@ protected boolean dispatchTask(TaskPriority taskPriority) {
...
                 addDispatchEvent(context, executionContext);
+            } else {
+                logger.info("Master failed to dispatch task to worker, taskInstanceId: {}", taskPriority.getTaskId());
             }
         } catch (RuntimeException e) {
-            logger.error("dispatch error: ", e);
+            logger.error("Master dispatch task to worker error: ", e);
         } catch (ExecuteException e) {
-            logger.error("dispatch error: {}", e.getMessage());
+            logger.error("Master dispatch task to worker error: {}", e.getMessage());
```
Comment (kezhenxu94) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/consumer/TaskPriorityQueueConsumer.java]: nit: also change to this so we can get the exception stack?


```suggestion
            logger.error("Master dispatch task to worker error: {}", e);
```

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskEvent.java [hunk 3]
```diff
@@ -27,10 +27,12 @@
 import java.util.Date;
 
 import io.netty.channel.Channel;
+import lombok.Data;
 
 /**
  * task event
  */
+@Data
```
Comment (kezhenxu94) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/TaskEvent.java:35]: 💯

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/ProcessInstance.java [hunk 4]
```diff
@@ -265,12 +265,13 @@ public ProcessInstance() {
      */
     public ProcessInstance(ProcessDefinition processDefinition) {
         this.processDefinition = processDefinition;
+        // todo: the name is not unique
```
Comment (kezhenxu94) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/ProcessInstance.java:268]: Can we do this in this PR?

**Review by kezhenxu94 — DISMISSED** [review]
Basically good to me. some nits

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/ProcessInstance.java (hunk 4 above)
Comment (ruanwenjun) [reply dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/ProcessInstance.java:268]: Right now, the `processInstanceName` is just used in front-end, we can change the `DateUtils.getCurrentTimeStamp()` to the snowflake id, but I am not sure if this is need to change, so I just add a todo in this PR.

**Review by kezhenxu94 — APPROVED** [review]
lgtm if all tests pass

**Pull request merged**